components/log-viewer-webui/client/src/components/QueryBox/index.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+    Input,
+    InputProps,
+    Progress,
+} from "antd";
+
+import {Nullable} from "../../typings/common";
+import styles from "./index.module.css";
+
+export interface QueryBoxProps extends InputProps {
+    /** The progress of the progress bar from `0` to `100`. Hide the bar if null. */
+    progress: Nullable<number>;
+}
+
+/**
+ * Renders an Input with a progress bar.
+ *
+ * @param queryBoxProps
+ * @return
+ */
+const QueryBox = (queryBoxProps: QueryBoxProps) => {
+    const inputProps = queryBoxProps as InputProps;
+    const {progress} = queryBoxProps;
```

Comment:
Let's destructure the props in the function prototype.
```suggestion
const QueryBox = ({
    progress,
    ...rest
}: QueryBoxProps) => {
```
then below we can write
```
            <Input {...rest}/>
```